app/services/promotional_tweet/post_promotional_tweet.rb

```diff
@@ -0,0 +1,42 @@
+class PostPromotionalTweet
+
+  def initialize(promotional_tweet, image, brand)
+    @message = promotional_tweet.message
+    @image   = image
+    @brand   = brand
+    @client  = brand.twitter_rest_client
+  end
+
+  def send!
+    @tweet = post_promo_tweet
+
+    if @tweet.present?
+      @promotional_tweet.update_attribute!(:tweet_url, @tweet.id)
+      tweet_url
+    else
+      raise StandardError.new('Failed to post to twitter')
+    end
+  end
+
+  private
+
+  def post_promo_tweet
+    if @image.present?
```

Comment:
Make these instance variables `attr_reader` so you don't need to call `@some_variable`.